server/compose/service/record.go

```diff
@@ -1279,6 +1284,38 @@ func (svc record) undelete(ctx context.Context, namespaceID, moduleID, recordID
 	undel.DeletedAt = nil
 	undel.DeletedBy = 0
 
+	// ensure module ref is set before running through records workflows and scripts
+	undel.SetModule(m)
+
+	undel.DeletedAt = nil
+	undel.DeletedBy = 0
+	undel.Revision = undel.Revision + 1
```

Comment:
Should the revision really change? I'm not sure so if it makes sense to you I won't argue